sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala

```diff
@@ -148,9 +148,9 @@ object ComputeCurrentTime extends Rule[LogicalPlan] {
 case class ReplaceCurrentLike(catalogManager: CatalogManager) extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = {
     import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
-    val currentNamespace = catalogManager.currentNamespace.quoted
-    val currentCatalog = catalogManager.currentCatalog.name()
-    val currentUser = CurrentUserContext.getCurrentUser
+    lazy val currentNamespace = catalogManager.currentNamespace.quoted
+    lazy val currentCatalog = catalogManager.currentCatalog.name()
+    lazy val currentUser = CurrentUserContext.getCurrentUser
```

Comment:
How about
```
    plan.transformAllExpressionsWithPruning(_.containsPattern(CURRENT_LIKE)) {
      case CurrentDatabase() =>
        Literal.create(catalogManager.currentNamespace.quoted, SQLConf.get.defaultStringType)
      case CurrentCatalog() =>
        Literal.create(catalogManager.currentCatalog.name(), SQLConf.get.defaultStringType)
      case CurrentUser() =>
        Literal.create(CurrentUserContext.getCurrentUser, SQLConf.get.defaultStringType)
    }
```
?

Reply:
We can remove these declaration.

Reply:
I think this improvement is not worth a JIRA ID. We can use [MINOR][SQL] instead.

Reply:
OK , i thought one line is a bit harder to read.  I moved the val to be in the cases, is it better?

Reply:
> I think this improvement is not worth a JIRA ID. We can use [MINOR][SQL] instead.

I changed the title, should I close the JIRA then?

Reply:
Yes. Please close it.